pkg/alertmanager/multitenant_test.go

```diff
@@ -0,0 +1,107 @@
+package alertmanager
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cortexproject/cortex/pkg/alertmanager/alerts"
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+)
+
+var (
+	simpleConfigOne = `route:
+  receiver: dummy
+
+receivers:
+  - name: dummy`
+
+	simpleConfigTwo = `route:
+  receiver: dummy
+
+receivers:
+  - name: dummy`
+)
+
+// basic easily configurable mock
+type mockAlertStore struct {
+	configs map[string]alerts.AlertConfigDesc
+}
+
+func (m *mockAlertStore) ListAlertConfigs(ctx context.Context) (map[string]alerts.AlertConfigDesc, error) {
+	return m.configs, nil
+}
+
+func TestMultitenantAlertmanager_loadAllConfigs(t *testing.T) {
+	mockStore := &mockAlertStore{
+		configs: map[string]alerts.AlertConfigDesc{
+			"user1": {
+				User:      "user1",
+				RawConfig: simpleConfigOne,
+				Templates: []*alerts.TemplateDesc{},
+			},
+			"user2": {
+				User:      "user2",
+				RawConfig: simpleConfigOne,
+				Templates: []*alerts.TemplateDesc{},
+			},
+		},
+	}
+
+	externalURL := flagext.URLValue{}
+	err := externalURL.Set("http://localhost/api/prom")
+	require.NoError(t, err)
+
+	tempDir, err := ioutil.TempDir(os.TempDir(), "alertmanager")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempDir)
+
+	am := &MultitenantAlertmanager{
+		cfg: &MultitenantAlertmanagerConfig{
+			ExternalURL: externalURL,
+			DataDir:     tempDir,
+		},
+		store:            mockStore,
+		cfgs:             map[string]alerts.AlertConfigDesc{},
+		alertmanagersMtx: sync.Mutex{},
+		alertmanagers:    map[string]*Alertmanager{},
+		stop:             make(chan struct{}),
+		done:             make(chan struct{}),
+	}
+
+	// Ensure the configs are synced correctly
+	err = am.updateConfigs()
```

Comment:
Can be compacted into `require.NoError(t, am.updateConfigs())`.

_Same comment applies below._